content/tutorials/publish-subscribe.textile

```diff
@@ -14,339 +14,7 @@ Messages published can contain string, JSON object, JSON array or binary data pa
-
-  "See this step in Github":https://github.com/ably/tutorials/commit/277475e
-
-blang[swift].
-  We will start by creating an Xcode project for this tutorial. To build your own Xcode Project in Swift visit "Apple developer website":https://developer.apple.com/library/content/documentation/IDEs/Conceptual/AppStoreDistributionTutorial/Setup/Setup.html and get familiar with steps necessary to setup your own application.
-  When you setup your application delete the default @ViewController.swift@ add new File -> New -> File... and choose @Cocoa Touch Class@.
-  <a href="/images/tutorials/tutorials-swift-CTC.png" target="_blank">
-    <img src="/images/tutorials/tutorials-swift-CTC.png" style="width: 100%" alt="Create new Cocoa Touch Class">
-  </a>
-  Name your new class "ExampleViewController" and choose @Swift@ as language:
-  <a href="/images/tutorials/tutorials-swift-ExampleClass.png" target="_blank">
-    <img src="/images/tutorials/tutorials-swift-ExampleClass.png" style="width: 100%" alt="Name new Cocoa Touch Class">
-  </a>
-  After that navigate to @Main.storyboard@ in your project, click on the @ViewController@ that has already been added by default during project creation and from the @Utilities@ that are located on the right choose @Identity Inspector@. Find the field labeled "Class" and select "ExampleViewController".
-  <a href="/images/tutorials/tutorials-swift-IB-class.png" target="_blank">
-    <img src="/images/tutorials/tutorials-swift-IB-class.png" style="width: 100%" alt="Interface design">
-  </a>
-
-  "See this step in Github":https://github.com/ably/tutorials/commit/a98f1bf
-
-  To start using Ably you first need to install the Ably pod via CocoaPods. You need to add a @Podfile@ to your project directory:
-
-  bc[sh]. touch Podfile
-
-  Then add this line to your application’s Podfile:
-
-  ```[text]
-    pod 'Ably'
-  ```
-
-  Install it with:
-
-  bc[sh]. pod install
-
-  To learn more about using CocoaPods in your project visit "official CocoaPods guide":https://guides.cocoapods.org/using/using-cocoapods.html.
-
-  Then in your files in which you will be using Ably import:
-
-  ```[swift]
-    import Ably
-  ```
-
-  To connect to Ably, you need to instance the client library with the API key you copied in Step 1. API keys used with "basic authentication":/general/authentication#basic-authentication for your own servers is generally preferred, however clients running on insecure devices should always use the "token authentication scheme":/general/authentication#token-authentication instead. In this example, we use an API key for simplicity.
-
-  Add the following to the file in which you imported the Ably library:
-
-  ```[swift]
-    let API_KEY = "INSERT-YOUR-API-KEY-HERE"
-    let client = ARTRealtime(key: API_KEY)
-  ```
-
-  "See this step in Github":https://github.com/ably/tutorials/commit/59dcc56
+<%= partial 'tutorials/_install_ablyrealtime', vars(why_realtime: "subscribe to messages") %>
```

Comment:
> I'm happy to expand this back out into duplicates, but TBH I'm a bit surprised to see you object to this, seemed like an obvious DRYing up

I refer you to https://go-proverbs.github.io/ and `A little copying is better than a little dependency.`

I love DRYing things up, but I objected to this specifically because:

* We have not to date used ERB to do conditional content in templates so we have now made it just a tad more complicated for new devs to work on these emplates
* There is no mechanism using `<%if%>` statements to work out if a block should be shown or not based on the languages supported as the page at that stage is not rendered and thus knowing which languages maybe further down in the page are declared is not possible.
* You had code like this `<% if @use_client_id %>As we'll be using presence, we also include the @clientId@, which is what we'll appear as in the presence set.<% end %>`, which means our DRY partial is now talking about presence, yet the var is `@use_client_id`.  I think the context of the setup will very quickly change per example and we'll probably end up with bugs if we go down this route.